### PR TITLE
nixos/tests/sway: init

### PR DESCRIPTION
**Pull request by primeos**
###### Motivation for this change

~~Needs some more love (WIP) but already works and IMO this would make a nice addition to e.g. also test Wayland, wlroots, and mesa.~~

~~I shamelessly stole the login function from `nixos/tests/ecryptfs.nix` btw. Could make sense to add an abstraction for that (haven't really looked at it so far - this was just a quick evening hack).~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by primeos** [comment]
@GrahamcOfBorg test sway

**Comment by Synthetica9** [comment]
> Needs some more love

Gave it some love https://github.com/primeos/nixpkgs/pull/4/files

File: nixos/tests/sway.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    machine.send_key("alt-3")
+    machine.succeed(
+        "su - alice -c 'swaymsg exec WINIT_UNIX_BACKEND=wayland DISPLAY=invalid alacritty'"
+    )
+    machine.wait_for_text("alice@machine")
+    machine.send_chars("wayland-info\n")
+    machine.sleep(5)
+    machine.screenshot("alacritty_wayland_info")
+    machine.send_key("alt-shift-q")
+    machine.wait_until_fails("pgrep alacritty")
```
Comment (Synthetica9) [inline nixos/tests/sway.nix:60]: This can be done with a `contextmanager`, but I'm not sure if I like that more or less...

Comment (primeos) [reply nixos/tests/sway.nix:60]: Avoiding redundancy is nice but in this case I don't think it's worth it (especially `contextmanager` and `**env` seem overkill but we could add a "normal" function later if it starts making sense).

Comment (primeos) [reply nixos/tests/sway.nix]: I kinda already oveengineered the test so maybe a normal function would make sense but for now I'd like to focus on getting an initial version merged and then we can always improve it later.

**Comment by primeos** [comment]
@GrahamcOfBorg test sway

File: nixos/tests/sway.nix (hunk 1 above)
Comment (Synthetica9) [reply nixos/tests/sway.nix]: > getting an initial version merged and then we can always improve it later.

Very true

**Comment by primeos** [comment]
Not sure what's up with the aarch64-linux timeout - it looks like the VM doesn't even start for some reason... Hopefully that only affects ofborg.

Let's give this test a go.

**Pull request merged**